tests/integration_tests/test_table_preprocessor.py

```diff
@@ -0,0 +1,110 @@
+
+from exasol_data_science_utils_python.preprocessing.encoding.ordinal_encoder import OrdinalEncoder
+from exasol_data_science_utils_python.preprocessing.normalization.min_max_scaler import MinMaxScaler
+from exasol_data_science_utils_python.preprocessing.schema.column import Column
+from exasol_data_science_utils_python.preprocessing.schema.schema import Schema
+from exasol_data_science_utils_python.preprocessing.schema.table import Table
+from exasol_data_science_utils_python.preprocessing.table_preprocessor import TablePreprocessor, \
+    ColumnPreprocesserDefinition
+
+
+def test_table_preprocessor_create_fit_queries():
+    c = pyexasol.connect(dsn="localhost:8888", user="sys", password="exasol")
+
+    c.execute("""
+    CREATE SCHEMA IF NOT EXISTS "SOURCE_SCHEMA"
+    """)
+    c.execute("""
+    CREATE SCHEMA IF NOT EXISTS "TARGET_SCHEMA"
+    """)
+    c.execute("""
+    CREATE OR REPLACE TABLE "SOURCE_SCHEMA"."SOURCE_TABLE" (
+        "CATEGORY" VARCHAR(1000),
+        "NUMERICAL" FLOAT
+    )
+    """)
+    c.execute("""INSERT INTO "SOURCE_SCHEMA"."SOURCE_TABLE" VALUES ('A',1),('B',2);""")
+    source_schema = Schema("SOURCE_SCHEMA")
+    source_table = Table("SOURCE_TABLE", source_schema)
+    target_schema = Schema("TARGET_SCHEMA")
+    source_column1 = Column("CATEGORY", source_table)
+    source_column2 = Column("NUMERICAL", source_table)
+    column_preprocessor_defintions = [
+        ColumnPreprocesserDefinition(source_column1.name, OrdinalEncoder()),
+        ColumnPreprocesserDefinition(source_column2.name, MinMaxScaler()),
+    ]
+
+    table_preprocessor = TablePreprocessor(target_schema, source_table, column_preprocessor_defintions)
+    queries = table_preprocessor.create_fit_queries()
+    for query in queries:
+        c.execute(query)
+
+    query = '''
+    SELECT "ID", "VALUE"
+    FROM "TARGET_SCHEMA"."SOURCE_SCHEMA_SOURCE_TABLE_CATEGORY_ORDINAL_ENCODER_DICTIONARY";
+     '''
+    result = c.execute(query).fetchall()
+    assert result == [(0, "A"), (1, "B")]
+
+    query = '''
+    SELECT "MIN", "RANGE"
+    FROM "TARGET_SCHEMA"."SOURCE_SCHEMA_SOURCE_TABLE_NUMERICAL_MIN_MAX_SCALAR_PARAMETERS";
+     '''
+    result = c.execute(query).fetchall()
+    assert result == [(1.0, 1.0)]
+
+    query = table_preprocessor.create_transform_query(source_table)
+    print(query)
+    c.execute(query)
+
+    query = '''SELECT * FROM "TARGET_SCHEMA"."SOURCE_SCHEMA_SOURCE_TABLE_TRANSFORMED"'''
+    result = c.execute(query).fetchall()
+    assert result == [(0, 0.0), (1, 1.0)]
+
+
+def test_table_preprocessor_transform_queries():
+    c = pyexasol.connect(dsn="localhost:8888", user="sys", password="exasol")
+
+    c.execute("""
+    CREATE SCHEMA IF NOT EXISTS "SOURCE_SCHEMA"
+    """)
+    c.execute("""
+    CREATE SCHEMA IF NOT EXISTS "TARGET_SCHEMA"
+    """)
+    c.execute("""
+    CREATE OR REPLACE TABLE "SOURCE_SCHEMA"."SOURCE_TABLE" (
+        "CATEGORY" VARCHAR(1000),
+        "NUMERICAL" FLOAT
+    )
+    """)
```

Comment:
This looks like a duplication

Reply:
yeah, I probably can refactor it out. However, in case of tests it sometimes better to keep duplicated, such that you don't need to jump around while understanding the test.